src/det.jl

```diff
@@ -21,6 +21,23 @@ end
     return vecdot(x0, cross(x1, x2))
 end
 
+@inline function _det(::Size{(4,4)}, A::StaticMatrix,S::Type)
+    A = similar_type(A,S)(A)
```

Comment:
yes there is a difference , it is not big (about 2 ns difference) , but the code produced if 
we avoid variable name changing type looks better .it looks like what you would expect  hand optimized assembly code would look like, a long list of vector intrinsics.